### PR TITLE
dash/dg2-credits

### PR DESCRIPTION
**Pull request by ddragula**
Added DataGrid credits.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208032830683340

### CODE REVIEW

File: ts/DataGrid/DataGrid.ts [hunk 1]
```diff
@@ -435,6 +453,13 @@ class DataGrid {
 
         let newDataTable = false;
         if (!this.dataTable || options.dataTable) {
+            this.userOptions.dataTable = options.dataTable;
+            if (this.options) {
+                this.options.dataTable = options.dataTable;
+            } else {
+                this.options = { dataTable: options.dataTable };
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid.ts:460]:
```
this.userOptions.dataTable = this.options?.dataTable || {};
this.options = { dataTable: options.dataTable };
```

Moreover, the `this.userOptions.dataTable` is not used anywhere, so what is the reason of this assignment?

File: ts/DataGrid/Options.ts [hunk 2]
```diff
@@ -392,25 +397,64 @@ export interface GroupedHeaderOptions {
...
+/**
+ * Options for the credits label.
+ */
+export interface CreditsOptions {
+    /**
+     * Whether to show the credits.
+     *
+     * @default true
+     */
+    enabled?: boolean;
```
Comment (sebastianbochan) [inline ts/DataGrid/Options.ts:433]: Why its optional, if its enabled by default?

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

File: ts/DataGrid/DataGrid.ts (hunk 1 above)
Comment (sebastianbochan) [reply ts/DataGrid/DataGrid.ts:460]: Suggestion: `(this.options || {}).dataTable = options.dataTable`

**Review by stitot — CHANGES_REQUESTED** [review]
Nice, but can we add target=_top to this link? Quite often datagrid will be in iframe, e.g. our own demos.

**Review by stitot — APPROVED** [review]

**Review by sebastianbochan — CHANGES_REQUESTED** [review]
IMO, credits should be disabled in Dashboards (DatagridComponent).

**Comment by ddragula** [comment]
> IMO, credits should be disabled in Dashboards (DatagridComponent) like it is in HC.

They are not disabled by default, they just have the credits manually disabled in the demos you are watching. Let's discuss this on daily.

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**